src/migratesettings.hpp

```diff
@@ -0,0 +1,14 @@
+// Minetest
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#include "settings.h"
+
+void migrate_settings()
+{
+	// Converts opaque_water to translucent_liquids
+	if (g_settings->existsLocal("opaque_water")) {
+		g_settings->set("translucent_liquids",
+				g_settings->getBool("opaque_water") ? "false" : "true");
+		g_settings->remove("opaque_water");
+	}
+}
```

Comment:
We don't use the `.hpp` file extension and code should not be in a header file.

Reply:
This is what  @Lars(appguru) came up with, as its not a standard header file. Its included that way because its only run one time. I advocated for doing it in a .cpp file and then having a header file but Lars thought it was to much for one function that will only be called once.

Reply:
I'll leave it to the other reviews to decide then.

Reply:
> This is why we chose .hpp as to refrence it as a code header not a cpp file or regular header

doesn't make much sense to me, just didn't notice the different file extension

Reply:
I can change it to .h but again its code too, i can just use .cpp and a sepreate .h but its only ever run one time. 


> > This is why we chose .hpp as to refrence it as a code header not a cpp file or regular header
> 
> doesn't make much sense to me, just didn't notice the different file extension


I can change to .h if you think its better

Reply:
I renamed to .h insted of .hpp

Reply:
> We don't use the `.hpp` file extension and code should not be in a header file.

This header is included exactly once, specifically in `main.cpp`, analogeous to `defaultsettings.h`, for a single function. There is no point in including it anywhere else. There will most certainly never be a point in including it in two different files.

`defaultsettings.cpp` has a corresponding header file, `defaultsettings.h`, which is also out of date. I do not see any benefit in the existence of `defaultsettings.h`.

It's just a function that's kept in a different file to not clutter `main.cpp`. As such, `.cpp` would be most appropriate, but including `.cpp` files is murky territory.

`.hpp` is the convention I use for "header-only" implementations such as this one. In our own sources, there is a weak precedent of this convention with `catch.hpp`.

I don't care about the file extension. But I do think that for this specific file there's no point in splitting it up in `.h` and `.cpp`.

Reply:
> > We don't use the `.hpp` file extension and code should not be in a header file.
> 
> This header is included exactly once, specifically in `main.cpp`, analogeous to `defaultsettings.h`, for a single function. There is no point in including it anywhere else. There will most certainly never be a point in including it in two different files.
> 
> `defaultsettings.cpp` has a corresponding header file, `defaultsettings.h`, which is also out of date. I do not see any benefit in the existence of `defaultsettings.h`.
> 
> It's just a function that's kept in a different file to not clutter `main.cpp`. As such, `.cpp` would be most appropriate, but including `.cpp` files is murky territory.
> 
> `.hpp` is the convention I use for "header-only" implementations such as this one. In our own sources, there is a weak precedent of this convention with `catch.hpp`. This convention seems to be relatively popular among "header-only" C++ libraries more generally (for example `json.hpp`).
> 
> I don't care about the file extension. But I do think that for this specific file there's no point in splitting it up in `.h` and `.cpp`.

I agree, there is no point in splitting it when its only called by main. So I changed the file name to .h as the others do not seem to think using .hpp is good.

Reply:
Seems fine to me. We have some header-only `.h` files already, though those are mostly templates, hence have no other choice.

I'll use `.h` for header-only files in Minetest in the future then.